components/skeleton/nz-skeleton.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { NzSkeletonAvatar, NzSkeletonParagraph, NzSkeletonTitle } from './nz-skeleton.type';
+
+@Component({
+  selector: 'nz-skeleton',
+  templateUrl: './nz-skeleton.component.html',
+  host: {
+    '[class.ant-skeleton]': 'true'
+  }
+})
+export class NzSkeletonComponent implements OnInit, OnChanges {
+  avartarClassMap;
+  private _title: NzSkeletonTitle = {};
+  private _avatar: NzSkeletonAvatar = {};
+  private _paragraph: NzSkeletonParagraph = {};
+  private _widthList: Array<number | string> = [];
```

Comment:
Private variables used in template file would cause an error in aot compiling.

Like: `[style.width]="toCSSUnit(_title.width)"`.